metrics/ci_integrations/test/source/jenkins/adapter/jenkins_source_client_adapter_test.dart

```diff
@@ -0,0 +1,597 @@
+
+void main() {
+  group("JenkinsSourceClientAdapter", () {
+    const jobName = 'test-job';
+    const defaultCoverage = Percent(0.6);
+    const defaultDuration = Duration(seconds: 10);
+    const defaultBuildUrl = 'buildUrl';
+    const defaultArtifact = JenkinsBuildArtifact(
+      fileName: 'coverage-summary.json',
+      relativePath: 'coverage/coverage-summary.json',
+    );
+    final defaultDateTime = DateTime(2020);
+
+    final jenkinsClientMock = _JenkinsClientMock();
+    final adapter = JenkinsSourceClientAdapter(jenkinsClientMock);
+    final responses = _JenkinsClientResponse(jobName);
+
+    PostExpectation<Future<InteractionResult>> whenFetchArtifact({
+      Matcher buildUrlThat,
+      Matcher relativePathThat,
+    }) {
+      return when(jenkinsClientMock.fetchArtifactByRelativePath(
+        argThat(buildUrlThat ?? anything),
+        argThat(relativePathThat ?? anything),
+      ));
+    }
+
+    PostExpectation<Future<InteractionResult>> whenFetchBuilds({
+      String jobName = jobName,
+      JenkinsQueryLimits limits,
+    }) {
+      whenFetchArtifact().thenAnswer(responses.artifact);
+      return when(jenkinsClientMock.fetchBuilds(
+        jobName,
+        limits: limits ?? anyNamed('limits'),
+      ));
+    }
+
+    /// Creates a [JenkinsBuild] instance with the given [buildNumber]
+    /// and default build arguments.
+    JenkinsBuild createJenkinsBuild({
+      @required int buildNumber,
+      JenkinsBuildResult result = JenkinsBuildResult.success,
+      bool building = false,
+      List<JenkinsBuildArtifact> artifacts = const [defaultArtifact],
+    }) {
+      return JenkinsBuild(
+        number: buildNumber,
+        duration: defaultDuration,
+        timestamp: defaultDateTime,
+        result: result,
+        url: defaultBuildUrl,
+        building: building,
+        artifacts: artifacts,
+      );
+    }
+
+    /// Creates a list of [JenkinsBuild] using the given [buildNumbers].
+    List<JenkinsBuild> createJenkinsBuilds({
+      @required List<int> buildNumbers,
+    }) {
+      return buildNumbers.map((buildNumber) {
+        return createJenkinsBuild(buildNumber: buildNumber);
+      }).toList();
+    }
+
+    /// Creates a [BuildData] instance with the given [buildNumber]
+    /// and default build arguments.
+    BuildData createBuildData({
+      @required int buildNumber,
+      BuildStatus buildStatus = BuildStatus.successful,
+    }) {
+      return BuildData(
+        buildNumber: buildNumber,
+        startedAt: defaultDateTime,
+        buildStatus: buildStatus,
+        duration: defaultDuration,
+        workflowName: jobName,
+        url: defaultBuildUrl,
+        coverage: defaultCoverage,
+      );
+    }
+
+    setUp(() {
+      reset(jenkinsClientMock);
+      responses.reset();
+    });
+
+    test(
+      ".fetchBuilds() fetches builds which are not building",
+      () {
+        final jenkinsBuilds = [
+          createJenkinsBuild(buildNumber: 1),
```

Comment:
```suggestion
          createJenkinsBuild(buildNumber: 1, building: false),
```
As it seems relevant to the test and not rely on the default value, readability should be improved as well with this change.